fs/glob.py

```diff
@@ -27,17 +36,87 @@
 )  # type: LRUCache[Tuple[Text, bool], Tuple[int, bool, Pattern]]
 
 
+def _split_pattern_by_rec(pattern):
```

Comment:
What does the `rec` in this function-name mean? I wonder if `_split_pattern_by_dir` or `_split_pattern_by_sep` would make more sense?